### DISCUSSION TITLE
Linq for Span<>, Memory<> with Pools

### DISCUSSION THREAD

**Discussion by DiegoStefanon** [dotnet/csharplang#6166] [Language Ideas]
Hi mi C#-Friends !
I have seen the  good work at  https://github.com/VictorNicollet/NoAlloq

I was wondering if it will be integrated, refined and extended in linq so as not to have to allocate memory all the time.

Many thanks !
Have a good time  and ENJOY !! ;-)

**Comment by huoyaoyuan** [comment]
Related discussion: #2482

**Comment by svick** [comment]
Looking at the performance results:

![](https://user-images.githubusercontent.com/287848/170465635-0d678487-8b81-4c2a-a015-6584bf9373d6.png)

This may be an impressive and useful library, but I think it's not good enough to add it into .Net as "high-performance LINQ", since it's still significantly slower than manual imperative code. Though fixing that would require runtime changes (especially related to allocations and inlining of lambdas), i.e. not something that can be done in a library.

**Reply by DiegoStefanon** [reply]
My friend
It Is not only about performances! but to avoid alsoa lot of memory allocation in conjunction with Memory<T> or ArrayPool ...
NoAlloq !

**Reply by DiegoStefanon** [reply]
C sharp has good Span<T> or Memory<T> and MemoryPool or ArrayPool, why not extend **Linq** also to use them ?
Look at **Allocated** column =>
![image](https://user-images.githubusercontent.com/7791527/170765196-d193c3ee-9260-4d71-ac5e-5db72b8ec2db.png)


On a large scale, allocations will lead to more frequent GC calls and performance decrease across all application layers.
